### ISSUE TITLE
Database entities should no be used directly in RestApi

### ISSUE THREAD

**Issue by svetoslavpetkov** [MilenIvanovv/Rental-Car-App#1]
All rest api endpoints should have their own objects. We should not use the db entities.

**Comment by svetoslavpetkov** [comment]
The purpose of making DTOs is no making the same structure as in the database 
The purpose of the DTO's is about hiding persistence from implementation

If I would be a consumer of a service for a car for example I would like to have the className and price in the car. By doing this we hide in the backend that the price is inherited by the class. This is a logic the rest api consumer should not care about.

Lets say in 3 months we want to change to have specific prices on a certain cars, that override the classPrice. -> with current approach it will need heavy changes on both backend and front end  

In short try again with DTO's and try to think of rest api (contract) without caring about how it is stored

**Comment by svetoslavpetkov** [comment]
Reason for separating entities is:
1. UI not to copy the persistence schema
2. database model not to be aware how it should be presented

This should not be done like this.

  @Expose()
  @Transform((carClass: CarClass, car) => car.class.price)
  price?: number;
